llap-server/src/java/org/apache/hadoop/hive/llap/io/encoded/OrcEncodedDataReader.java

```diff
@@ -276,11 +275,20 @@ public void unpause() {
 
   @Override
   protected Void callInternal() throws IOException, InterruptedException {
-    return ugi.doAs(new PrivilegedExceptionAction<Void>() {
-      @Override
-      public Void run() throws Exception {
-        return performDataRead();
+    return ugi.doAs((PrivilegedExceptionAction<Void>) () -> {
+      long startTime = counters.startTimeCounter();
+      try {
+        performDataRead();
+        consumer.setDone();
+      } catch (Throwable t) {
+        consumer.setError(t);
```

Comment:
reuse `handleReaderError`

Reply:
I deleted handleReaderError, this is the only place all handling happens.

Reply:
ok